src/BlazorWebView/src/Maui/Android/WebKitWebViewClient.cs

```diff
@@ -107,14 +106,19 @@ public override void OnPageFinished(AWebView? view, string? url)
 
 		private void RunBlazorStartupScripts(AWebView view)
 		{
-			// TODO: we need to protect against double initialization because the
-			// OnPageFinished event refires after the app is brought back from the 
-			// foreground and the webview is brought back into view, without it actually
-			// getting reloaded.
-
-
-			// Set up JS ports
+			// Confirm Blazor hasn't already initialized
 			view.EvaluateJavascript(@"
+				(function() { return typeof(window.__BlazorStarted); })();
```

Comment:
Ideally we'd use the `started` property, but it's lost to minification: 
https://cs.github.com/dotnet/aspnetcore/blob/d4e09edefac2978b114030d55eefde22361accfd/src/Components/Web.JS/src/Boot.WebView.ts?q=Blazor+has+already+started#L12